### ISSUE TITLE
WinAppDriver cannot find visible objects

### ISSUE THREAD

**Issue by Alyaan24** [microsoft/WinAppDriver#1592]
I have been facing a really weird issue, I've seen other people face it as well but haven't seen a clear fix, my tests sometimes pass, and sometimes randomly fail with the error saying it couldnt find an element with those search parameters. I've ran the same tests together and all 16 of them passed and then ran them again the next morning and they all failed. sometimes they barely pass and sometimes they all do, I've been stuck on this for a long while and would appreciate any help. This is the error log on a simple test to check if a toggle button is turned on, it detects the settings button usually and presses it but then fails to recognize the toggle button.

Windows Application Driver listening for requests at: http://127.0.0.1:4723/
Press ENTER to exit.

==========================================
POST /session HTTP/1.1
Accept: application/json, image/png
Connection: Keep-Alive
Content-Length: 247
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.0 (.net windows)

HTTP/1.1 200 OK
Content-Length: 163
Content-Type: application/json

{"sessionId":"BFA5FB1C-D162-4F21-90E6-5D6715D40BC0","status":0,"value":{"app":"App pathway","ms:waitForAppLaunch":5,"platformName":"Windows"}}

==========================================
POST /session/BFA5FB1C-D162-4F21-90E6-5D6715D40BC0/timeouts HTTP/1.1
Accept: application/json, image/png
Content-Length: 31
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.0 (.net windows)

{"type":"implicit","ms":5000.0}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"BFA5FB1C-D162-4F21-90E6-5D6715D40BC0","status":0}

==========================================
POST /session HTTP/1.1
Accept: application/json, image/png
Content-Length: 170
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.0 (.net windows)

{"desiredCapabilities":{"app":"Root","deviceName":"WindowsPC","newCommandTimeout":90,"platformName":"Windows"},"capabilities":{"firstMatch":[{"platformName":"Windows"}]}}
HTTP/1.1 200 OK
Content-Length: 111
Content-Type: application/json

{"sessionId":"A65EA81E-B0E5-4DD0-9F20-231510190D05","status":0,"value":{"app":"Root","platformName":"Windows"}}

==========================================
POST /session/BFA5FB1C-D162-4F21-90E6-5D6715D40BC0/element HTTP/1.1
Accept: application/json, image/png
Content-Length: 35
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.0 (.net windows)

{"using":"name","value":"Settings"}
HTTP/1.1 200 OK
Content-Length: 101
Content-Type: application/json

{"sessionId":"BFA5FB1C-D162-4F21-90E6-5D6715D40BC0","status":0,"value":{"ELEMENT":"42.788248.4.428"}}

==========================================

POST /session/BFA5FB1C-D162-4F21-90E6-5D6715D40BC0/element/42.788248.4.428/click HTTP/1.1
Accept: application/json, image/png
Content-Length: 2
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.0 (.net windows)

{}
HTTP/1.1 200 OK
Content-Length: 63
Content-Type: application/json

{"sessionId":"BFA5FB1C-D162-4F21-90E6-5D6715D40BC0","status":0}

==========================================
POST /session/BFA5FB1C-D162-4F21-90E6-5D6715D40BC0/element HTTP/1.1
Accept: application/json, image/png
Content-Length: 45
Content-Type: application/json;charset=utf-8
Host: 127.0.0.1:4723
User-Agent: selenium/3.141.0 (.net windows)

{"using":"class name","value":"ToggleSwitch"}
HTTP/1.1 404 Not Found
Content-Length: 139
Content-Type: application/json

{"status":7,"value":{"error":"no such element","message":"An element could not be located on the page using the given search parameters."}}

Let me know if you need any more info

**Comment by anunay1** [comment]
"They all failed" because of the same issue "An element could not be located on the page using the given search parameters"

**Comment by Alyaan24** [comment]
> "They all failed" because of the same issue "An element could not be located on the page using the given search parameters"

Yes but clearly the elements i'm trying to find are on the screen and the names are correct

**Comment by anunay1** [comment]
Can you get the page source when they all failed?

**Comment by Alyaan24** [comment]
> Can you get the page source when they all failed?

Im sorry, I'm pretty new to this, what do you mean by page source?

**Comment by Shakevg** [comment]
> > Can you get the page source when they all failed?
> 
> Im sorry, I'm pretty new to this, what do you mean by page source?

 
@Alyaan24 This is content of the page in XML format that extracted by WinAppDriver and used to search items https://github.com/microsoft/WinAppDriver/search?q=pagesource
